src/copyCodeBlock.js

```diff
@@ -2,31 +2,57 @@ import { getDisplayString, getClipboardString } from './utils';
 import styles from './styles';
 
 export default (string, opts = {}) => {
-    const lang = opts && opts.lang || '';
+    const { lang = 'HTML', shouldReturnDomEl = false } = opts;
     const { container, displayCode, copyButton } = styles(opts);
     const buttonText = 'Click to copy';
 
+    const onclick = btnId => `(function() {
+        var textarea = document.createElement('textarea');
+        var button = document.querySelector('#${btnId}');
+
+        textarea.innerHTML = '${getClipboardString(string)}';
+        document.body.appendChild(textarea);
+        textarea.select();
+
+        button.innerHTML = 'Copied!';
+        setTimeout(function() { button.innerHTML = '${buttonText}'; }, 1500);
+
+        document.execCommand('copy');
+        document.body.removeChild(textarea);
+    })();`;
+
+    if (shouldReturnDomEl) {
```

Comment:
We don't have to have another code path here that constructs the DOM by hand; we can reuse the string below by doing:
```javascript
const domContent = `...stuff below...`;

const div = document.createElement('div');
div.innerHTML = domContent;
return div.children[0];

Reply:
Ahh, tried that but forgot the `.children[0]`